### PR TITLE
feat(assignment): add permissions

### PR DESCRIPTION
**Pull request by bartnaessens**
Add permission EDIT_OWN_ASSIGNMENT
Teachers can no longer access the edit page of an other teacher's assignment

### CODE REVIEW

File: src/authentication/helpers/permission-names.ts [hunk 1]
```diff
@@ -11,6 +11,8 @@ export enum PermissionName {
 	PUBLISH_ALL_COLLECTIONS = 'PUBLISH_ALL_COLLECTIONS',
 	PUBLISH_OWN_COLLECTIONS = 'PUBLISH_OWN_COLLECTIONS',
 	CREATE_ASSIGNMENTS = 'CREATE_ASSIGNMENTS',
+	EDIT_ALL_ASSIGNMENTS = 'EDIT_ALL_ASSIGNMENTS',
+	EDIT_OWN_ASSIGNMENTS = 'EDIT_OWN_ASSIGNMENTS',
 	EDIT_ASSIGNMENTS = 'EDIT_ASSIGNMENTS',
```
Comment (bertyhell) [inline src/authentication/helpers/permission-names.ts:16]: Add comment:
```
EDIT_ASSIGNMENTS = 'EDIT_ASSIGNMENTS', // DEPRECATED use EDIT_OWN_ASSIGNMENTS instead
```

**Review by bertyhell — CHANGES_REQUESTED** [review]

File: src/authentication/helpers/permission-names.ts (hunk 1 above)
Comment (bartnaessens) [reply src/authentication/helpers/permission-names.ts]: fixed

File: src/assignment/views/AssignmentOverview.tsx [hunk 2]
```diff
@@ -112,13 +112,16 @@ const AssignmentOverview: FunctionComponent<AssignmentOverviewProps> = ({
 		try {
 			if (user) {
 				setCanEditAssignments(
-					await PermissionService.hasPermissions(PermissionName.EDIT_ASSIGNMENTS, user)
+					await PermissionService.hasPermissions(
+						[PermissionName.EDIT_OWN_ASSIGNMENTS, PermissionName.EDIT_ASSIGNMENTS],
```
Comment (bertyhell) [inline src/assignment/views/AssignmentOverview.tsx:116]: shouldn't this also include the new permission: EDIT_ALL_ASSIGNMENTS ?

File: src/assignment/views/AssignmentEdit.tsx [hunk 3]
```diff
@@ -89,131 +88,111 @@ const AssignmentEdit: FunctionComponent<DefaultSecureRouteProps<{ id: string }>>
...
+					[
+						PermissionName.EDIT_ALL_ASSIGNMENTS,
+						{ name: PermissionName.EDIT_ASSIGNMENTS, obj: tempAssignment },
+						{ name: PermissionName.EDIT_OWN_ASSIGNMENTS, obj: tempAssignment },
+					],
+					user
+				))
+			) {
+				history.push(`/${ROUTE_PARTS.assignments}/${assignmentId}`);
+				ToastService.danger(
```
Comment (bertyhell) [inline src/assignment/views/AssignmentEdit.tsx:144]: ToastService.info seems more appropriate. Since we mostly use danger for stuff related to deletes

**Review by bertyhell — CHANGES_REQUESTED** [review]

File: src/assignment/views/AssignmentOverview.tsx (hunk 2 above)
Comment (bartnaessens) [reply src/assignment/views/AssignmentOverview.tsx]: indeed. fixed

File: src/assignment/views/AssignmentEdit.tsx (hunk 3 above)
Comment (bartnaessens) [reply src/assignment/views/AssignmentEdit.tsx]: updated.

**Review by bertyhell — APPROVED** [review]

**Pull request merged**